### PR TITLE
Dashboard app deploy icon

### PR DESCRIPTION
**Pull request by charlesfleche**
From the Dashboard, to reach the Deploy page of an application one has to first click on the app name, wait for the page to load, then click on Deploy. This PR adds an icon linked to the deploy page next to an application name. To avoid visual clutter this icon is only visible when the mouse hover over an application name.

dashboard_app_deploy_icon

These patches add a new apps_paginated_tile.html template (used by a new appsPaginatedTileDirective dashboard directive). A new TileType.APPS_PAGINATE now specializes the AppsPaginatedContext model.

Possible caveats :
- dashboardDirectives.appsPaginatedTileDirective and apps_paginated_tile.html come from brutals copy / paste. Some code refactoring may be necessary if the number of specializations starts to grow
- applications are not displayed using the default bootstap list-group because a link markup must not have interactive content (like another link) descendant. See http://stackoverflow.com/questions/9882916/are-you-allowed-to-nest-a-link-inside-of-a-link and http://www.w3.org/html/wg/drafts/html/master/single-page.html#the-a-element. In the standard bootstrap list-group an entire item can be an link markup : hovering the mouse over a line puts it in "link / hand" mode. But adding a clickable icon would violate the "no nested link" rule. In this PR, one has to click on the app name. Still, hovering in the item white space activates the grey background
- app and dashboard pages have slightly different icons as different versions of Awesome fonts seems to be used for these pages

### CODE REVIEW

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by czue** [comment]
nice!

File: corehq/apps/dashboard/static/dashboard/ng_partials/apps_paginated_tile.html [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+               title="">{{ title }}</a>
+        </div>
+        <div class="panel-body">
+            <i class="dashboard-icon dashboard-icon-medium dashboard-icon-bg {{ default.icon }}"
+               ng-show="showItemList()"></i>
+            <i class="fa fa-spinner fa-spin fa-5x" ng-show="showSpinner()"></i>
+            <div class="list-group" ng-show="showItemList()">
+                <div class="list-group-item" data-ng-repeat="item in paginatedItems" >
+                   <a href="{{ item.url }}">{{ item.name }}</a>
+                   <a class="icon" href="{{ item.release_manager_url }}"><i class="icon-alternate-download fa fa-download"></i></a>
```
Comment (biyeun) [inline corehq/apps/dashboard/static/dashboard/ng_partials/apps_paginated_tile.html]: i wouldn't call this class `icon`, as that has historic meaning from Bootstrap 2 styles and will certainly cause confusion. Perhaps add a modifier like `secondary-link`. Also `icon-alternate-download` should not be used as the icon class, as we are not using glyphicons, but fontawesome

File: corehq/apps/dashboard/static/dashboard/js/hq_dashboard.ng.js [hunk 2]
```diff
@@ -180,6 +180,20 @@
         };
     };
 
+    dashboardDirectives.appsPaginatedTileDirective = function (dashboardConfig) {
```
Comment (biyeun) [inline corehq/apps/dashboard/static/dashboard/js/hq_dashboard.ng.js]: can you think of a way to do this without adding a specific directive for applications? Making a secondary link option generic could potentially be useful for other paginated lists. Allowing the icon class to be specified for the secondary link would help with this.

**Comment by biyeun** [comment]
@charlesfleche this is a great idea! just want to make sure this is implemented in a more generic way so that it can be useful for future tiles. :-)

File: corehq/apps/dashboard/static/dashboard/ng_partials/apps_paginated_tile.html (hunk 1 above)
Comment (charlesfleche) [inline corehq/apps/dashboard/static/dashboard/ng_partials/apps_paginated_tile.html]: That makes sense, I'll fix that.

**Comment by dimagimon** [comment]
:+1: No lint errors found.

File: corehq/apps/dashboard/static/dashboard/ng_partials/apps_paginated_tile.html (hunk 1 above)
Comment (charlesfleche) [inline corehq/apps/dashboard/static/dashboard/ng_partials/apps_paginated_tile.html]: icon has been renamed  in a6e37e0
Reference to glyphicons has been deleted in 1673fca

**Comment by charlesfleche** [comment]
Sure. I'll try to push a more generic version ASAP.

File: corehq/apps/dashboard/models.py [hunk 3]
```diff
@@ -307,13 +321,25 @@ def applications(self):
...
+
+        def _get_app_name(app):
+            return app['key'][1]
+
         apps = self.applications[self.skip:self.skip + self.limit]
-        return [{
-            'name': a['key'][1],
-            'url': _get_app_url(a),
-        } for a in apps]
+
```
Comment (dimagimon) [inline corehq/apps/dashboard/models.py]: W293 blank line contains whitespace

File: corehq/apps/dashboard/models.py [hunk 4]
```diff
@@ -307,13 +321,25 @@ def applications(self):
...
         apps = self.applications[self.skip:self.skip + self.limit]
-        return [{
-            'name': a['key'][1],
-            'url': _get_app_url(a),
-        } for a in apps]
+        
+        return [self._fmt_item(_get_app_name(a),
+                               _get_app_url(a),
+                               None, # description
```
Comment (dimagimon) [inline corehq/apps/dashboard/models.py]: E261 at least two spaces before inline comment

File: corehq/apps/dashboard/models.py [hunk 5]
```diff
@@ -307,13 +321,25 @@ def applications(self):
...
+                               None, # full_name
```
Comment (dimagimon) [inline corehq/apps/dashboard/models.py]: E261 at least two spaces before inline comment

**Comment by charlesfleche** [comment]
bd96784 was supposed to make this list display more generic… In this version there only one pagination template. It receives an item. Depending on the values of item.secondary_url and item.secondary_item_icon, an icon is added (or not) at the end of each lilne.

It almost work, until I found that angularjs doesn't allow condition in its template. It what I tried to do here, with a {% if item.secondary_url and item.secondaray_url_icon %}{% else %}{% endif %} block.

I'm very neophyte when it comes to web development and right now I wonder what would the best way to continue this branch. It seems like paginated_tile.html should first be used in angularjs, then another template within paginated_tile.html should be selected depending on the secondary_url* values.

What do you guys think ?

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by charlesfleche** [comment]
Error timed out while getting http://files.dimagi.com/pygooglechart-0.3.0.tar.gz

File: corehq/apps/dashboard/static/dashboard/ng_partials/paginated_tile.html [hunk 6]
```diff
@@ -14,17 +14,33 @@
...
-                   data-ng-repeat="item in paginatedItems"
-                   popover-placement="right"
-                   popover-title="{{ item.name_full }}"
-                   popover="{{ item.description }}"
-                   popover-trigger="mouseenter"
-                   track-analytics=""
-                   title="">
-                    {{ item.name }}
-                </a>
+                {% if item.secondary_url and item.secondary_url_icon %}
```
Comment (biyeun) [inline corehq/apps/dashboard/static/dashboard/ng_partials/paginated_tile.html]: Be careful with this, as angular templates may share similar syntax with django templates but ultimately they are two different things. See the `ng-if` directive for what you'd like to do here: https://docs.angularjs.org/api/ng/directive/ngIf

**Comment by dimagimon** [comment]
:+1: No lint errors found.

**Comment by biyeun** [comment]
looks good. waiting for build

**Comment by czue** [comment]
thanks @charlesfleche !

**Pull request merged**